sympy/stats/crv.py

```diff
@@ -340,10 +340,11 @@ def probability(self, condition, **kwargs):
             # Integrate out all other random variables
             pdf = self.compute_density(rv, **kwargs)
             # return S.Zero if `domain` is empty set
-            if domain.set is S.EmptySet:
+            domain_set = domain.set
```

Comment:
Why introduce a variable for this? domain.set does not do anything time-costly, it just returns `domain.args[1]`. There shouldn't be any noticeable overhead of using domain.set three times in this code, especially compared to Integral below.

Reply:
I introduced the variable because I thought that instead of calling the method thrice, it would be better to store the value. You are right, compared to the integral, the time is negligible. I have updated the PR accordingly.